crates/runtime/src/lib.rs

```diff
@@ -647,13 +645,21 @@ impl Runtime {
                     Ok(accelerator) => accelerator,
                     Err(err) => {
                         let ds_name = &ds.name;
-                        status::update_dataset(ds_name, status::ComponentStatus::Error);
+                        self.status
+                            .update_dataset(ds_name, status::ComponentStatus::Error);
                         metrics::datasets::LOAD_ERROR.add(1, &[]);
                         warn_spaced!(spaced_tracer, "{} {err}", ds_name.table());
                         continue;
                     }
                 };
 
+                // If we already have an existing file, it means there is data from a previous acceleration and we don't need
+                // to wait for the first refresh to complete to mark it ready.
+                if accelerator.has_existing_file(ds) {
+                    self.status
+                        .update_dataset(&ds.name, status::ComponentStatus::Ready);
+                }
```

Comment:
This is the check for an existing acceleration file

Reply:
@phillipleblanc what if there's an existing file but the previous initial load did not complete? It seems like there needs to be an additional check that the file exists and at least the initial load completed. You could read this from task history/metrics, but it would be evicted. Perhaps one way is to store a semaphore such as `.spice/data/ready.sem`.

Reply:
I will handle this in a follow-on PR.